test/dynamo/test_export.py

```diff
@@ -2219,6 +2225,9 @@ def my_dyn_fn(a, b, c):
 
         torch._dynamo.export(my_dyn_fn, x, y, z)
         constraints = [dynamic_dim(x, 0), dynamic_dim(y, 0), dynamic_dim(z, 0)]
+        with self.assertRaises(ConstraintViolationError):
+            torch._dynamo.export(my_dyn_fn, x, y, z, constraints=constraints)
+        constraints.append(dynamic_dim(z, 0) == dynamic_dim(x, 0))
```

Comment:
I guess I will get more clarity as I keep reading the diff. Just to ask, dynamic dim equality means two dims must be both dynamic and have the same range values, correct?

Reply:
Not just. They must be the same value. Usually this will be because there's a guard that forces them to be the same (e.g., they are compared equal, tensors with those dimensions are added together, etc.)

In contrast, 4 and 5 both satisfy the same range 3 < s < 6 (but are not equal).